File: app/schemas/hack.py
"""Pydantic schemas for the Hacks feature.

Defines request and response models used by the FastAPI endpoints. Response
models enable ORM mode to populate from SQLAlchemy objects.
"""

from pydantic import BaseModel, Field, ConfigDict
from typing import Optional, List
from datetime import datetime


class HackBase(BaseModel):
    """Fields shared by create and response models."""

    # Concise headline
    title: str = Field(..., min_length=1, max_length=200)

    # Main body text of the article/tip
    content: str = Field(..., min_length=1, max_length=20000)

    # Optional grouping label (e.g. productivity, wellness)
    category: Optional[str] = Field(None, max_length=50)

    # Optional tags list presented to clients; stored as CSV in DB
    tags: Optional[List[str]] = None


class HackCreate(HackBase):
    """Payload for creating a hack."""

    pass


class HackUpdate(BaseModel):
    """Partial update payload; only provided fields are updated."""

    title: Optional[str] = Field(None, min_length=1, max_length=200)
    content: Optional[str] = Field(None, min_length=1, max_length=20000)
    category: Optional[str] = Field(None, max_length=50)
    tags: Optional[List[str]] = None


class HackResponse(BaseModel):
    """Response model returned to clients."""

    id: int
    title: str
    content: str
    category: Optional[str] = None
    tags: Optional[List[str]] = None
    created_at: datetime
    updated_at: datetime

    # Enable reading from SQLAlchemy objects
    model_config = ConfigDict(from_attributes=True)


class HackListResponse(BaseModel):
    """Standard list wrapper with pagination counters."""

    success: bool = True
    data: List[HackResponse]
    total: int
    limit: int
    offset: int


class HackSingleResponse(BaseModel):
    """Standard single-entity wrapper."""

    success: bool = True
    data: HackResponse

File: android/app/src/main/java/com/moodmate/data/model/Hack.kt
package com.moodmate.data.model

data class HackCreate(
    val title: String,
    val content: String,
    val category: String? = null,
    val tags: List<String>? = null
)

data class HackUpdate(
    val title: String? = null,
    val content: String? = null,
    val category: String? = null,
    val tags: List<String>? = null
)

data class HackResponse(
    val id: Int,
    val title: String,
    val content: String,
    val category: String? = null,
    val tags: List<String>? = null,
    val created_at: String,
    val updated_at: String
)

data class HackListResponse(
    val success: Boolean,
    val data: List<HackResponse>,
    val total: Int,
    val limit: Int,
    val offset: Int
)

data class HackSingleResponse(
    val success: Boolean,
    val data: HackResponse
)

data class WellnessTip(
    val id: Int,
    val title: String,
    val description: String,
    val category: String? = null,
    val tags: List<String>? = null
)

